src/appimagetool.c

```diff
@@ -903,6 +911,28 @@ main (int argc, char *argv[])
                         printf("Will not guess update information since zsyncmake is missing\n");
                     }
                 }
+	    } else if(GITHUB_REPOSITORY){
+		gchar *zsyncmake_path = g_find_program_in_path ("zsyncmake");
+		if (zsyncmake_path){
+		   char buf[1024];
+		   printf("Running on GitHub Actions\n");
+		   if (strstr(GITHUB_REF, "/pull/")) {
+			printf("Will not calculate update information for GitHub because this is a pull request\n");
+		   } else {
+                        gchar **parts = g_strsplit (GITHUB_REPOSITORY, "/", 0);
+
+			char* channel;
+			if (GITHUB_REF != "" && strstr(GITHUB_REF, "/continuous/")) {
+				channel = "latest";
+			} else {
+				channel = "continuous";
+			}
+			sprintf(buf, "gh-releases-zsync|%s|%s|%s|%s*-%s.AppImage.zsync", parts[0], parts[1], channel, app_name_for_filename, arch);
```

Comment:
No boundary checks. Potential buffer overflow. Use `snprintf` and implement proper error handling to be sure to notice overflows and handle them correctly.

Reply:
Done on https://github.com/AppImage/AppImageKit/pull/1075/commits/63b6e6a7df944eeb1a1198762ea353f141e0061c